spec/models/knight_spec.rb

```diff
@@ -1,95 +1,103 @@
 require 'rails_helper'
 
 RSpec.describe Knight, type: :model do
-  describe 'white knight movement validation' do
-    it 'should return false if not moved' do
-      knight = FactoryGirl.create(:knight, :white)
-      expect(knight.valid_move?(1, 7)).to eq false
-    end
-
-    it 'should return false if moved 2 spaces right' do
-      knight = FactoryGirl.create(:knight, :white)
-      expect(knight.valid_move?(3, 7)).to eq false
-    end
-
-    it 'should return false if moved 1 space forward' do
-      knight = FactoryGirl.create(:knight, :white)
-      expect(knight.valid_move?(1, 6)).to eq false
-    end
-
-    it 'should return false if moved 3 spaces forward and 3 spaces right' do
-      knight = FactoryGirl.create(:knight, :white)
-      expect(knight.valid_move?(4, 4)).to eq false
-    end
+  let(:game) { FactoryGirl.create(:game) }
+  let(:knight) do
+    game.pieces.find_by(
+      type: 'Knight',
+      color: 'white',
+      x_position: 6,
+      y_position: 7
+    )
+  end
+  let(:moved_knight) do
+    game.pieces.create(
+      type: 'Knight',
+      color: 'white',
+      x_position: 3,
+      y_position: 3,
+      moved: true
+    )
+  end
 
-    it 'should return true if moved 2 spaces forward and 1 space left' do
-      knight = FactoryGirl.create(:knight, :white)
-      expect(knight.valid_move?(2, 5)).to eq true
+  describe 'creation' do
+    it 'should create a white knight' do
+      knight = FactoryGirl.create(:knight, color: 'white')
+      expect(knight.type).to eq('Knight')
     end
 
-    it 'should return true if moved to 2 spaces right and 1 space forward' do
-      knight = FactoryGirl.create(:knight, :white)
-      expect(knight.valid_move?(3, 6)).to eq true
+    it 'should fail to create a red knight' do
+      expect { FactoryGirl.create(:knight, color: 'red') }.to\
+        raise_error(ActiveRecord::RecordInvalid)
     end
   end
 
-  describe 'black knight movement validation' do
+  describe 'moved' do
     it 'should return false if not moved' do
-      knight = FactoryGirl.create(:knight, :black)
-      expect(knight.valid_move?(1, 0)).to eq false
-    end
-
-    it 'should return false if moved 2 spaces right' do
-      knight = FactoryGirl.create(:knight, :black)
-      expect(knight.valid_move?(3, 0)).to eq false
+      expect(knight.move!(6, 7)).to eq false
+      expect(knight.x_position).to eq 6
+      expect(knight.y_position).to eq 7
+      expect(knight.moved).to eq false
     end
+  end
 
-    it 'should return false if moved 1 space forward' do
-      knight = FactoryGirl.create(:knight, :black)
-      expect(knight.valid_move?(1, 1)).to eq false
+  describe 'invalid moveset' do
```

Comment:
If you mean whether or not a piece is being moved from its original location, that is being tested here:  `  describe 'moved' do
    it 'should return false if not moved' do
      expect(knight.move!(6, 7)).to eq false
      expect(knight.x_position).to eq 6
      expect(knight.y_position).to eq 7
      expect(knight.moved).to eq false
    end
  end`